File: hub/src/light/effect/effect_modifier.py
from light.effect.resolved_effect import ResolvedEffect
from light.effect.partial_effect import PartialEffect
from light.effect.effect_priority import EffectPriority

class BasicAmplitudeModifier(PartialEffect):
    def __init__(self):
        super().__init__(0)
        self.isModifier = True

File: hub/drumlite-js/src/light/effect/EffectModifier.ts
import PartialEffect, { EffectParameters } from "./PartialEffect";

export default class EffectModifier<T extends EffectParameters> extends PartialEffect<T> {
    public isModifier: boolean;
    
    constructor(params: T) {
        super(params, 0);
        this.isModifier = true;
    }
}
